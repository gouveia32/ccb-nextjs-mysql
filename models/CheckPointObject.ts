export interface CheckPointInterface {
    id?: number | string;
    text: string;
    checked: boolean;
  }
  
  export type CheckPointType = CheckPointInterface;
  
  export const CheckPointObject: CheckPointType = {
           id: 0,
           text: "",
           checked: false,
         };
  
  export enum cCheckPoint {
    id = 'id',
    text = 'text',
    checked = 'checked',
  }