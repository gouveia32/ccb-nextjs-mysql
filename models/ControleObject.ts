export interface ControleInterface {
  id?: number | string;
  texto: string;
  marcado: boolean;
}

export type TipoControle = ControleInterface;

export const ControleObject: TipoControle = {
         id: 0,
         texto: "",
         marcado: false,
       };

export enum cControle {
  id = 'id',
  texto = 'texto',
  marcado = 'marcado',
}
