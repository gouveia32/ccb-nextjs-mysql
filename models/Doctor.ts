export interface DoctorModel {
  id: string;
  name: string;
  email: string;
  image?: string;
}

export type DoctorType = DoctorModel;

export const cDoctorModel = {
  id: "id",
  name: "name",
  email: "email",
  image: '',
};

export const DoctorObject: DoctorType = {
  id: "",
  name: "",
  email: "",
  image: '',
};

export function isDoctorType(object: any): object is DoctorType {
  return "name" in object;
}
