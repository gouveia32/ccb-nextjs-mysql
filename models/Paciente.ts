export interface PacienteModel {
  id: string;
  nome: string;
}

export type TipoPaciente = PacienteModel;

export const cPacienteModel = {
  id: "id",
  nome: "nome"
};

export const PacienteObject: TipoPaciente = {
  id: "",
  nome: "",
};

export function isPacienteType(object: any): object is TipoPaciente {
  return "nome" in object;
}
