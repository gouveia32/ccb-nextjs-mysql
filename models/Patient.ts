export interface PatientModel {
  id: string;
  name: string;
}

export type PatientType = PatientModel;

export const cPatientModel = {
  id: "id",
  name: "name"
};

export const PatientObject: PatientType = {
  id: "",
  name: "",
};

export function isPatientType(object: any): object is PatientType {
  return "name" in object;
}
