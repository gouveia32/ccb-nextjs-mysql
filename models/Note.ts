import { TagType } from "./Tag";
import { CheckPointType } from "./CheckPointObject";
import { Note } from "@prisma/client";

export enum NoteTypeEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NoteModel {
  id?: string;
  name: string;
  noteType: NoteTypeEnum;
  content: string;
  color: string;
  pinned: boolean;
  createdAt: string;
  tags: TagType[];
  checkPoints?: CheckPointType[];
}

export type NoteType = NoteModel;

export const NoteObject: NoteType = {
  id: "",
  name: "",
  noteType: NoteTypeEnum.TEXT,
  content: "",
  color: "#fff",
  pinned: false,
  createdAt: new Date(Date.now()).toString(),
  tags: [],
  checkPoints: [],
};

export const NoteDBObject: Note = {
  id: "",
  name: "",
  noteType: "TEXT",
  color: "",
  content: "",
  pinned: false,
  createdAt: new Date(Date.now()),
  doctorId: "",
};

export enum cNoteModel {
  id = "id",
  name = "name",
  noteType = "noteType",
  color = "color",
  content = "content",
  pinned = "pinned",
  createdAt = "createdAt",
  tags = "tags",
  checkPoints = "checkPoints",
}
