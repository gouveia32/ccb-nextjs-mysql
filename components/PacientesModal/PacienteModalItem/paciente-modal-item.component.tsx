import React, { useCallback, useEffect, useRef, useState } from "react";
import { PacienteType } from "../../../models/Paciente";
import { Grid, IconButton, TextField } from "@material-ui/core";
import LabelOutlinedIcon from "@material-ui/icons/LabelOutlined";
import DeleteOutlineOutlinedIcon from "@material-ui/icons/DeleteOutlineOutlined";
import CheckOutlinedIcon from "@material-ui/icons/CheckOutlined";

export interface PacienteModalItemProps {
  paciente: PacienteType;
  onDeletePaciente: (paciente: PacienteType) => void;
  onUpdatePaciente: (paciente: PacienteType) => void;
}

const PacienteModalItem: React.FC<PacienteModalItemProps> = ({
  paciente,
  onDeletePaciente,
  onUpdatePaciente,
}: PacienteModalItemProps) => {
  const [edit, setEdit] = useState(false);

  const [updatePaciente, setUpdatePaciente] = useState(paciente);

  const pacienteItemRef = useRef(null);

  const clickOutsideListener = useCallback(
    (e: MouseEvent) => {
      if (
        pacienteItemRef.current &&
        !(pacienteItemRef.current as any).contains(e.target)
      ) {
        setEdit(false);
      }
    },
    [pacienteItemRef]
  );

  useEffect(() => {
    document.addEventListener("click", clickOutsideListener);
    return () => {
      document.removeEventListener("click", clickOutsideListener);
    };
  }, [clickOutsideListener]);

  const handleOnChange = (value: string) => {
    setUpdatePaciente((prevState) => {
      const newPaciente = Object.assign({}, prevState);
      newPaciente.name = value;
      return newPaciente;
    });
  };

  const renderItems = (
    <Grid
      container={true}
      className="py-2 cursor-pointer"
      onClick={(e) => {
        setEdit(true);
        e.stopPropagation();
      }}
    >
      <Grid item={true}>
        <LabelOutlinedIcon />
      </Grid>
      <Grid item={true} style={{ flex: 1 }}>
        <h5 className="m-0 ms-2">{paciente.name}</h5>
      </Grid>
      <Grid item={true}>
        <IconButton
          size={"small"}
          onClick={(event) => {
            onDeletePaciente(paciente);
            event.stopPropagation();
          }}
        >
          <DeleteOutlineOutlinedIcon />
        </IconButton>
      </Grid>
    </Grid>
  );

  const renderEditItems = (
    <Grid container={true} className="py-2">
      <Grid item={true}>
        <IconButton size={"small"} onClick={() => onDeletePaciente(paciente)}>
          <DeleteOutlineOutlinedIcon />
        </IconButton>
      </Grid>
      <Grid item={true} style={{ flex: 1 }}>
        <TextField
          value={updatePaciente.name}
          fullWidth={true}
          size={"small"}
          variant={"standard"}
          onChange={(event) => handleOnChange(event.target.value)}
          onKeyDown={(event) =>
            event.code === "Enter" && onUpdatePaciente(updatePaciente)
          }
        />
      </Grid>
      <Grid item={true}>
        <IconButton size={"small"} onClick={() => onUpdatePaciente(updatePaciente)}>
          <CheckOutlinedIcon />
        </IconButton>
      </Grid>
    </Grid>
  );

  return <div ref={pacienteItemRef}>{edit ? renderEditItems : renderItems}</div>;
};

export default React.memo(PacienteModalItem);
