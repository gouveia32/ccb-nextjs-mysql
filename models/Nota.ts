import { TipoPaciente } from "./Paciente";
import { TipoControle } from "./ControleObject";
import { Nota } from "@prisma/client";

export enum tipoNotaEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NotaModel {
  id?: string;
  nome: string;
  tipoNota: tipoNotaEnum;
  conteudo: string;
  cor: string;
  pino: boolean;
  criadoEm: string;
  pacientes: TipoPaciente[];
  controles?: TipoControle[];
}

export type TipoNota = NotaModel;

export const NotaObject: TipoNota = {
  id: "",
  nome: "",
  tipoNota: tipoNotaEnum.TEXT,
  conteudo: "",
  cor: "#fff",
  pino: false,
  criadoEm: new Date(Date.now()).toString(),
  pacientes: [],
  controles: [],
};

export const NotaDBObject: Nota = {
  id: "",
  nome: "",
  tipoNota: "TEXT",
  cor: "",
  conteudo: "",
  pino: false,
  criadoEm: new Date(Date.now()),
  medicoId: "",
};

export enum cNotaModel {
  id = "id",
  nome = "nome",
  tipoNota = "tipoNota",
  cor = "cor",
  conteudo = "conteudo",
  pino = "pino",
  criadoEm = "criadoEm",
  pacientes = "pacientes",
  controles = "controles",
}
