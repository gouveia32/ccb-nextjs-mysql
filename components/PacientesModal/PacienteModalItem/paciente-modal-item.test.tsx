import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import PacienteModalItem, { PacienteModalItemProps } from "./paciente-modal-item.component";
import { PacienteObject } from "../../../models/Paciente";
import { IconButton } from "@material-ui/core";

describe("PacienteModalItem component", () => {
  let wrapper: ShallowWrapper;
  let mockOnUpdatePaciente: any;
  let mockOnDeletePaciente: any;

  beforeEach(() => {
    mockOnUpdatePaciente = jest.fn();
    mockOnDeletePaciente = jest.fn();

    const mockProps: PacienteModalItemProps = {
      paciente: PacienteObject,
      onUpdatePaciente: mockOnUpdatePaciente,
      onDeletePaciente: mockOnDeletePaciente,
    };

    wrapper = shallow(<PacienteModalItem {...mockProps} />);
  });

  it("should render PacienteModalItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onDelete when IconButton component is clicked", () => {
    wrapper
      .find(IconButton)
      .simulate("click", { stopPropagation: () => undefined });
    expect(mockOnDeletePaciente).toHaveBeenCalled();
  });
});
