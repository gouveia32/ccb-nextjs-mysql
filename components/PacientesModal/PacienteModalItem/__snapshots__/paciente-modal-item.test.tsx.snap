// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacienteModalItem component should render PacienteModalItem component 1`] = `ShallowWrapper {}`;
